server/server.go

```diff
@@ -3,52 +3,214 @@ package server
+
+	if len(output) <= 0 {
+		output += `"Upload Status":"Successfully Uploaded File(s)"`
+	}
+
+	response, err := json.Marshal(output)
+	if err != nil {
+		log.Println(err)
+	}
+
+	// Write the response to be sent to client.
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	w.Write(response)
+}
+
+// buildRun builds and runs the assignment uploaded.
+func build(w http.ResponseWriter, r *http.Request) {
 
 	w.Header().Set("Access-Control-Allow-Origin", "*")
 
-	responseString := readFormData(r)
+	var outputString string
+	var currDir string
+	var err error
+
+	// Navigate to the assignment working directory.
+	outputString, currDir = navigateToWorkDir()
+	if outputString == "" {
+		// Execute the compile command.
+		outputString, err = runCommand(assignTestingInfo.CommandToCompile)
+		if err != nil {
+			log.Println("error while building the assignment", err)
+			outputString = err.Error()
+		}
 
-	if len(responseString) <= 0 {
-		responseString += `"Upload Status":"Successfully Uploaded File(s)"`
+		// Navigate back to the code-runner working directory after successful execution.
+		err = os.Chdir(currDir)
+		if err != nil {
+			log.Println("error while navigating to the current directory", err)
+		}
 	}
 
-	responseS, err := json.Marshal(responseString)
+	if err == nil {
+		outputString = "Compiled successfully"
+	}
+	response, err := json.Marshal(outputString)
 	if err != nil {
 		log.Println(err)
 	}
+	// Write the response to be sent to client.
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	w.Write(response)
+}
+
+// buildRun builds and runs the assignment uploaded.
+func run(w http.ResponseWriter, r *http.Request) {
+
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+
+	var outputString string
+	var currDir string
+	var err error
 
+	// Navigate to the assignment working directory.
+	outputString, currDir = navigateToWorkDir()
+	if outputString == "" {
+
+		// Append the command line arguments to run command.
+		runCmd := assignTestingInfo.CommandToExecute
+		for _, value := range assignTestingInfo.CmdlineArgs {
+			runCmd = fmt.Sprintf("%s %s", runCmd, value)
+		}
+		// Execute the assignment run command.
+		outputString, err = runCommand(runCmd)
+		if err != nil {
+			log.Println("error while executing the assignment", err)
+			outputString = err.Error()
+		}
+
+		// Navigate back to the code-runner working directory after successful execution.
+		err = os.Chdir(currDir)
+		if err != nil {
+			log.Println("error while navigating to the current directory", err)
+		}
+	}
+
+	response, err := json.Marshal(outputString)
+	if err != nil {
+		log.Println(err)
+	}
 	// Write the response to be sent to client.
 	w.Header().Set("Content-Type", "application/json")
 	w.WriteHeader(http.StatusOK)
-	w.Write(responseS)
+	w.Write(response)
+}
+
+// navigateToWorkDir navigates to the provided working directory of the assignment.
+func navigateToWorkDir() (string, string) {
```

Comment:
I feel this could be `func navigateToWorkDir() (string, error)` similar to `runCommand(...)`